### PR TITLE
Symfony 4 support

### PR DESCRIPTION
**Pull request by JonEastman**
Added Symfony 4 support. 

#SymfonyConHackday2017

### CODE REVIEW

**Comment by dbu** [comment]
thanks! can you please also add the test to the .travis.yml matrix to run tests with symfony 3.4 and 4.0 as explained in http://symfony.com/blog/helping-prepare-for-symfony-4-bundle-support ? right now the only build is with symfony 3.3 so we don't know if things actually work.

File: .travis.yml [hunk 1]
```diff
@@ -19,12 +19,21 @@ matrix:
       env: TRANSPORT=jackrabbit COMPOSER_FLAGS="--prefer-lowest"
     - php: 7.1
       env: TRANSPORT=doctrine_dbal SYMFONY_VERSION="3.3" PHPBENCH="yes"
+    - php: 7.1
+      env: TRANSPORT=doctrine_dbal SYMFONY_VERSION="3.4" PHPBENCH="yes"
```
Comment (ElectricMaxxx) [inline .travis.yml]: I think you schould use `^3.4` or `3.4.*`, can you try that?

Comment (JonEastman) [reply .travis.yml]: @ElectricMaxxx  Yeah, I'll give it a try!

**Comment by ElectricMaxxx** [comment]
@JonEastman seems to work. Composer runs on this build, you should do same for all.

**Comment by ElectricMaxxx** [comment]
here is the reason: https://github.com/phpcr/phpcr-utils/blob/master/composer.json#L32

**Comment by ElectricMaxxx** [comment]
I created: https://github.com/phpcr/phpcr-utils/pull/177/files

**Comment by JonEastman** [comment]
@ElectricMaxxx Sounds good!  I'll update the the Google doc .

**Comment by ElectricMaxxx** [comment]
Now you need latest phpcr-utils. @dbu @lsmith77 release or 1.4@dev?

**Comment by dbu** [comment]
i tagged an utils version, will restart the build now. (and then go visit the dracula castle in bran, in case you never hear from me again ;-) )

**Comment by ElectricMaxxx** [comment]
ok :-)

@dbu now it is your (companies) packages: https://github.com/liip/RMT/pull/129

**Comment by ElectricMaxxx** [comment]
I restarted here, lets see which package is next to fix :-)

**Comment by dbu** [comment]
@JonEastman can you please change the requirement on liip/rmt to `2.6.*@dev` then we can see if things work out or there is another issue after that.

File: composer.json [hunk 2]
```diff
@@ -21,16 +21,16 @@
...
-        "symfony/console": "^3.3"
+        "symfony/console": "^3.3 || ^4.0"
     },
     "require-dev": {
-        "symfony/yaml": "^3.3",
-        "symfony/phpunit-bridge": "^3.2",
-        "liip/rmt": "~1.2",
+        "symfony/yaml": "^3.3 || ^4.0",
+        "symfony/phpunit-bridge": "^3.2 || ^4.0",
+        "liip/rmt": "2.6.*@dev",
```
Comment (ElectricMaxxx) [inline composer.json]: You should use 1.2.* or ^1.2

**Comment by JonEastman** [comment]
@dbu and @ElectricMaxxx I updated the liip/rmt to 1.2.*   @dbu I wasn't sure what you meant about updating the liip/rmt to 2.6.*@dev?  First time contributing at SymfonyCon so I'm still trying to figure out how this all works!

**Comment by ElectricMaxxx** [comment]
No worries. You made a great job.

**Comment by ElectricMaxxx** [comment]
I do not know, which is the correct version, i just saw, that there is no version with 2.* at least not at packagist. I will look now.

**Comment by ElectricMaxxx** [comment]
I can not find any new release @dbu?

File: composer.json [hunk 3]
```diff
@@ -21,16 +21,16 @@
...
-        "symfony/console": "^3.3"
+        "symfony/console": "^3.3 || ^4.0"
     },
     "require-dev": {
-        "symfony/yaml": "^3.3",
-        "symfony/phpunit-bridge": "^3.2",
-        "liip/rmt": "~1.2",
+        "symfony/yaml": "^3.3 || ^4.0",
+        "symfony/phpunit-bridge": "^3.2 || ^4.0",
+        "liip/rmt": "1.2.*",
```
Comment (dbu) [inline composer.json]: sorry, i seem to have been confused yesterday. no idea where i found the 2. whatever... liip/RMT had no branch-alias, so the thing did not work. i now added that. can you please change this to `1.3.*@dev`? this time it should work, packagist sees the 1.3 version.

**Comment by ElectricMaxxx** [comment]
@JonEastman anything we can help now?

**Comment by ElectricMaxxx** [comment]
Now it is `phpbench` which is symfony 4.0 compatible on master. I have no clue which of our dependencies requires it, @dbu you?

**Comment by ElectricMaxxx** [comment]
ahh it is directly in `.travis.yml` @JonEastman can you increase that one on `1.0@dev`, please. I hope we got it then :-)

**Comment by ElectricMaxxx** [comment]
Yippy ... :-) @dbu  i think we got it.

**Comment by JonEastman** [comment]
Awesome!  Thanks for the walk through @dbu and @ElectricMaxxx

**Comment by ElectricMaxxx** [comment]
@dbu can you restart the `phpcr-bundle` build, when this is merged. i think we use `2.0@dev` there, so no tagging should be needed here – i hope.

**Comment by dbu** [comment]
awesome, thanks a ton for your persistence @JonEastman !

**Comment by Tobion** [comment]
Would have been nice if this got applied to the 1.4 branch as 2.0 is not released yet.

**Comment by dbu** [comment]
looks like we did not actually need to change anything, so should be easy to backport. do you need it for something specific @Tobion ?

**Comment by Tobion** [comment]
Not really. I just wondered why the FosElasticaBundle tests didn't install some some sf 4 components.

**Comment by dbu** [comment]
ah, ok. trying the backport in #772 as i don't want to rush the release of 2.0 of phpcr-odm and there seem to be no changes needed for the library.

**Pull request merged**